universal-login-web3/lib/ULWeb3Provider.ts

```diff
@@ -3,41 +3,57 @@ import {Config, getConfigForNetwork, Network} from './config';
 import UniversalLoginSDK, {WalletService} from '@universal-login/sdk';
 import {MetamaskService} from './services/MetamaskService';
 import {UIController} from './services/UIController';
-import {initUi} from './ui';
 import {providers, utils} from 'ethers';
 import {Callback, JsonRPCRequest, JsonRPCResponse} from './models/rpc';
-import {ensure, Message} from '@universal-login/commons';
+import {ensure, Message, walletFromBrain} from '@universal-login/commons';
 import {waitFor} from './utils';
+import {initUi} from './ui';
+import {AppProps} from './ui/App';
+import {StorageService, WalletStorageService} from '@universal-login/react';
+
+export interface ULWeb3ProviderOptions {
+  provider: Provider;
+  relayerUrl: string;
+  ensDomains: string[];
+  uiInitializer?: (services: AppProps) => void;
+  storageService?: StorageService;
+}
```

Comment:
I generally recommend to split options into two types:

- common options (ie provider, relayerUrl, ensDomains) and call them Config Those options are usually plain models.
- testing/customization options (ie uiInitializer, storageService) and call them 
overrides
Those options are usually services - an alternative to common choices in dependency injection